### PR TITLE
drivers/kw2xrf/kw2xrf.c: fix bug and cleanup in _assemble_tx_buf

### PR DESCRIPTION
**Pull request by jfischer-phytec-iot**
BUG: the pktbuf was released at the wrong place and
return value of _assemble_tx_buf was not verified.

### CODE REVIEW

File: drivers/kw2xrf/kw2xrf.c [hunk 1]
```diff
@@ -1064,30 +1064,26 @@ void kw2xrf_isr_event(gnrc_netdev_t *netdev, uint32_t event_type)
...
-     */
-    /*if (dev->option & KW2XRF_OPT_AUTOACK) {
-        dev->buf[1] = 0x61;
-    }
-    else {
-        dev->buf[1] = 0x51;
-    }*/
-    dev->buf[1] = 0x51;
+    /* we are building a data frame here */
+    dev->buf[1] = IEEE802154_FCF_TYPE_DATA;
```
Comment (PeterKietzmann) [inline drivers/kw2xrf/kw2xrf.c]: It's unrelated to this PR but why do you start for write the frame at buffer position `[1]` and not at `[0]`?

File: drivers/kw2xrf/kw2xrf.c [hunk 2]
```diff
@@ -1064,30 +1064,26 @@ void kw2xrf_isr_event(gnrc_netdev_t *netdev, uint32_t event_type)
...
+    //dev->buf[1] = 0x51;
```
Comment (PeterKietzmann) [inline drivers/kw2xrf/kw2xrf.c]: If  you need that as a reminder, please make a "real" comment out of it, describing what it is good for

File: drivers/kw2xrf/kw2xrf.c [hunk 3]
```diff
@@ -1118,19 +1114,16 @@ int _assemble_tx_buf(kw2xrf_t *dev, gnrc_pktsnip_t *pkt)
...
 
     /* fill in source PAN ID (if applicable */
-    /* FIXME: compare with at86rf2xx, currently not working
     if (dev->option & KW2XRF_OPT_USE_SRC_PAN) {
         dev->buf[index++] = (uint8_t)((dev->radio_pan) & 0xff);
         dev->buf[index++] = (uint8_t)((dev->radio_pan) >> 8);
     } else {
         dev->buf[1] |= IEEE802154_FCF_PAN_COMP;
     }
-    */
```
Comment (PeterKietzmann) [inline drivers/kw2xrf/kw2xrf.c]: Would you spend a newline for the `else` in L:1124 :-) ?

**Comment by PeterKietzmann** [comment]
Running exmples/gnrc_networking on a samr21-xpro and a pba-d-01-kw2x and sniffing the traffic showed that there is an automatic retransmission of four by default on the atmel board on no automatic retransmission on the phytec board. Is that intentional by the hardware? Have not time to look into this... Anyway, the other stuff looks fine. Please quickly address my minor comments. Will merge then

**Comment by OlegHahm** [comment]
@PeterKietzmann, thanks for the review!

File: drivers/kw2xrf/kw2xrf.c (hunk 1 above)
Comment (jfischer-phytec-iot) [inline drivers/kw2xrf/kw2xrf.c]: [0] contains the size of packet
https://github.com/RIOT-OS/RIOT/pull/4264/files#diff-cae64f326f164c9b37208e6a9f3f9c8dR1211

The driver is currently being rebuilt. I will make it more understandable.

File: drivers/kw2xrf/kw2xrf.c (hunk 2 above)
Comment (jfischer-phytec-iot) [inline drivers/kw2xrf/kw2xrf.c]: no, it should be deleted

File: drivers/kw2xrf/kw2xrf.c (hunk 3 above)
Comment (jfischer-phytec-iot) [inline drivers/kw2xrf/kw2xrf.c]: yes

**Comment by jfischer-phytec-iot** [comment]
>Running exmples/gnrc_networking on a samr21-xpro and a pba-d-01-kw2x and sniffing the traffic showed that there is an automatic retransmission of four by default on the atmel board on no automatic retransmission on the phytec board. Is that intentional by the hardware? Have not time to look into this... 

We did the same thing and know it :-). It will be fixed in another PR, the driver is currently being rebuilt.

**Comment by PeterKietzmann** [comment]
Okay, so please squash then

**Comment by PeterKietzmann** [comment]
ACK and go

**Pull request merged**